controllers/component_build_controller.go

```diff
@@ -194,177 +195,207 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+			}
+			// Nothing to do
+			return ctrl.Result{}, nil
+		}
+
+		log.Info("Starting Pipelines as Code provision for the Component")
+
+		if err := r.ProvisionPaCForComponent(ctx, &component); err != nil {
+			log.Error(err, "Pipelines as Code provision for the Component failed")
+			return ctrl.Result{}, err
+		}
+		log.Info("Pipelines as Code provision for the Component finished successfully")
+
+		// Update PaC annotation to show that PaC provision is done
+		if err := r.Client.Get(ctx, req.NamespacedName, &component); err != nil {
+			log.Error(err, "failed to get Component")
+			return ctrl.Result{}, err
+		}
+		if len(component.Annotations) == 0 {
+			component.Annotations = make(map[string]string)
+		}
+		component.Annotations[PaCProvisionAnnotationName] = PaCProvisionDoneAnnotationValue
+		if err := r.Client.Update(ctx, &component); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	// Pipelines as Code workflow is not enabled, use plain builds.
+
+	// Check initial build annotation to know if any work should be done for the Component
 	if len(component.Annotations) == 0 {
 		component.Annotations = make(map[string]string)
 	}
-	if component.Annotations[InitialBuildAnnotationName] == "true" {
+	if _, exists := component.Annotations[InitialBuildAnnotationName]; exists {
 		// Initial build have already happend, nothing to do.
 		return ctrl.Result{}, nil
 	}
-	// The initial build annotation is absent, onboarding of the component needed
-
-	gitopsConfig := gitopsprepare.PrepareGitopsConfig(ctx, r.Client, component)
-	if val, ok := component.Annotations[gitops.PaCAnnotation]; (ok && val == "1") || gitopsConfig.IsHACBS {
-		// Use pipelines as code build
-		log.Info("Pipelines as Code enabled")
-
-		gitProvider, err := gitops.GetGitProvider(component)
-		if err != nil {
-			log.Error(err, "error detecting git provider")
-			// Do not reconcile, because configuration must be fixed before it is possible to proceed.
-			return ctrl.Result{}, nil
-		}
+	// The initial build is needed for the Component
 
-		// Expect that the secret contains token for Pipelines as Code webhook configuration,
-		// but under <git-provider>.token field. For example: github.token
-		// Also it can contain github-private-key and github-application-id
-		// in case GitHub Application is used instead of webhook.
-		pacSecret := corev1.Secret{}
-		pacSecretKey := types.NamespacedName{Namespace: component.Namespace, Name: gitopsprepare.PipelinesAsCodeSecretName}
-		if err := r.Client.Get(ctx, pacSecretKey, &pacSecret); err != nil {
-			if !errors.IsNotFound(err) {
-				log.Error(err, fmt.Sprintf("failed to get Pipelines as Code secret in %s namespace", component.Namespace))
-				r.EventRecorder.Event(&pacSecret, "Warning", "ErrorReadingPaCSecret", err.Error())
-				return ctrl.Result{}, err
-			}
+	// Set initial build annotation to prevent next builds
+	component.Annotations[InitialBuildAnnotationName] = "processed"
```

Comment:
should the annotation value be a constant as well?

Reply:
It could, but the value are ignored for now and only presence of the annotation matters.